fecgov/fec-cms
# Format resource page for court cases

## Summary

- Resolves #4191
- Add `formatted-title` option to resource pages
- Remove top-border on resource page sections when there is no intro-text to avoid double-border appearance
- By changing `{{ block.sections}}` -TO- `{% include_block block.sections %}` , we get access to context in the child partial template so if we want to in the future we can conditionally apply styling or template structure for **court cases only** using something like `{% if 'court-cases' in request.path %}`

**Add-ons to this PR:**
- Add on commit to change wording on JFC participants, fix for recently merged [PR] (https://github.com/fecgov/fec-cms/pull/5304)
- Add-on commit to Include missing migration that appears in CF but not on GH repos. 
     - The extra migration is related to the fact that two PRs made changes to the same thing (ResourcePage - sections) at the same time ([#5094](https://github.com/fecgov/fec-cms/pull/5094)  AND [#5096](https://github.com/fecgov/fec-cms/pull/5096) ) and so the merge of the two (0123_merge_20220308_1116.py) did not take, requiring one more `makemigration`s before everything is copasetic.

### Required reviewers

one frontend, one UX, one content (to make sure ResourcePage sections still work as they want it)

## Impacted areas of the application

General components of the application that this PR will affect:
	modified:   fec/static/scss/components/_options.scss
	modified:   fec/templates/long_page.html
	new file:   home/migrations/0124_auto_20220706_0036.py
	modified:   home/models.py
	modified:   home/templates/blocks/section.html
	modified:   home/templates/home/resource_page.html

## Screenshots

<img width="1225" alt="Screen Shot 2022-07-06 at 12 38 59 AM" src="https://user-images.githubusercontent.com/5572856/177471920-90c663f0-c4f4-4704-a9fe-a91dde84ba23.png">


## How to test

- [x] Dont merge before feature deploy task is removed
- `./manage.py migrate`
- `npm run build`
- Test court case pages to format title with italics or any HTML
- Check other existing Resource Pages to make sure that the top border on the first section does not show up when there is no intro text
- Confirm that you can hide titles and add another as an H3 in the rich-text field or leave it without a title
    - The top border should be there only if there is intro-text added
